opensearch_py_ml/semantic search/train.py

```diff
@@ -0,0 +1,362 @@
+# SPDX-License-Identifier: Apache-2.0
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+# Any modifications Copyright OpenSearch Contributors. See
+# GitHub history for details.
+
+#  Licensed to Elasticsearch B.V. under one or more contributor
+#  license agreements. See the NOTICE file distributed with
+#  this work for additional information regarding copyright
+#  ownership. Elasticsearch B.V. licenses this file to you under
+#  the Apache License, Version 2.0 (the "License"); you may
+#  not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+# 	http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing,
+#  software distributed under the License is distributed on an
+#  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+#  KIND, either express or implied.  See the License for the
+#  specific language governing permissions and limitations
+#  under the License.
+
+import argparse
+import os
+import pickle
+import random
+import shutil
+import time
+from zipfile import ZipFile
+
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+import torch
+from sentence_transformers import InputExample, SentenceTransformer
+from tqdm import tqdm
+from transformers import get_linear_schedule_with_warmup
+
+
+def train_semantic_search_model(
+    read_path: str = None,
+    model_url: str = None,
+    output_model_name: str = None,
+    output_model_path: str = None,
+    zip_file_name: str = None,
+) -> None:
+    """
+    Description:
+    read the synthetic queries and use it to fine tune/train a sentence transformer model to save a zip file
+
+    Parameters:
+    read_path: str
+        Optional, path to read the generated queries zip file, if None, default as 'synthetic_query' folder in current directory
+    model_url: str = None
+        the url to download sentence transformer model, if None, default as 'sentence-transformers/msmarco-distilbert-base-tas-b'
+    output_model_path: str=None #MSC
+        the path to store trained custom model. If None, default as current folder path
+    output_model_name: str=None #MSC
+        the name of the trained custom model. If None, default as 'trained_model.pt'
+    zip_file_name: str =None
+        Optional, file name for zip file. if None, default as zip_model.zip
+    Return:
+        None
+    """
+
+    query_df = read_queries(read_path)
+    train_examples = load_sentence_transformer_example(query_df)
+    train_model(train_examples, model_url, output_model_path, output_model_name)
+    zip_model(output_model_path, zip_file_name)
+    return None
+
+
+#     helpful functions:
+
+
+def read_queries(read_path: str = None) -> pd.DataFrame:
```

Comment:
it's not assigned any read path to default value, then the default is None. It used to later if the user didn't change default which is None, it will raise Exception that user didn't provide a read_path.